finality-aleph/src/new_network/service.rs

```diff
@@ -126,20 +126,29 @@ impl<N: Network, D: Data> Service<N, D> {
         }
     }
 
-    fn next_sender(&mut self) -> Option<NotificationSender> {
+    async fn send(network: &N, send_queue: &mut VecDeque<(D, PeerId, Protocol)>) -> Option<()> {
```

Comment:
`Option<()>` is a _really_ strange return type, and you are not even using the `?` trick from the previous version which would have justified it at least a little bit.

The most expressive variant of this I can think of would return `Option<Result<(), NetworkSendError>>`, preferably with `NetworkSendError` containing the peer id and a way of getting printed that could be useful in the`tokio::select!` for printing. This way you could make the inside of the loop something like
```
let (data, peer, protocol) = send_queue.front()?;
let result = network.send(data.encode(), *peer, protocol.name()).await;
send_queue.pop_front();
return Some(result);
```
or even better not make it a loop at all, making everything simpler. You lose the `trace` about an empty queue, but it's not that useful anyway.

Reply:
Done in this way